src/components/trainings/trainingCard3.component.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import {
+  Image,
+  View,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ImageProps,
+} from 'react-native';
+import { Exercise } from '@src/core/model/exercise.model';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import { Button } from '@kitten/ui';
+import { Text } from '@src/components/common';
+import {
+  ClockIcon,
+  FlashIcon,
+} from '@src/assets/icons';
+
+interface ComponentProps {
+  training: Exercise;
+  index?: number;
+  onDetails: (index: number) => void;
+  onTiming: (index: number) => void;
+  onEnergy: (index: number) => void;
+}
+
+export type TrainingCardProps = ThemedComponentProps & ComponentProps & TouchableOpacityProps;
+
+class TrainingCard3Component extends React.Component<TrainingCardProps> {
+
+  private onDetails = (): void => {
+    this.props.onDetails(this.props.index);
+  };
+
+  private onTiming = (): void => {
+    this.props.onTiming(this.props.index);
+  };
+
+  private onEnergy = (): void => {
+    this.props.onEnergy(this.props.index);
+  };
+
+  private renderTimingIcon = (): React.ReactElement<ImageProps> => {
+    return ClockIcon(this.props.themedStyle.timingIcon);
+  };
+
+  private renderEnergyIcon = (): React.ReactElement<ImageProps> => {
```

Comment:
You forgot about `style` argument, which means style provided by `eva` is not applied to component. Also `eva` style usually contains `width` and `height`, so you don't need it in StyleSheet

Reply:
Nice catch but I've just wanted to override size of the icon. Moreover there are some icon margins, that I don't need in such particular case, from 'eva'-ghost-icon applied. So we can look at this exam like on customization example.